### PR TITLE
Fixed #26126 -- Fixed transient failure of test_max_age_expiration

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/26126

### CODE REVIEW

**Comment by timgraham** [comment]
I think 994d6137a2519436d17d5b3d16cb49a3fa79f93e would help. Sorry I forget about that when accepting the ticket.

**Comment by berkerpeksag** [comment]
Good idea. I've updated the PR, thanks!

File: tests/requests/tests.py [hunk 1]
```diff
@@ -246,11 +246,14 @@ def test_far_expiration(self):
 
     def test_max_age_expiration(self):
         "Cookie will expire if max_age is provided"
-        response = HttpResponse()
-        response.set_cookie('max_age', max_age=10)
+        current_time = time.time()
+        with freeze_time(current_time):
+            response = HttpResponse()
+            response.set_cookie('max_age', max_age=10)
```
Comment (timgraham) [inline tests/requests/tests.py]: I think only `set_cookie()` needs to be within the context manager, correct?

Comment (berkerpeksag) [inline tests/requests/tests.py]: Technically, yes, but it looks a bit weird:

```py
def test_foo(self):
    current_time = time.time()
    response = HttpResponse()
    with freeze_time(current_time):
        response.set_cookie('max_age', max_age=10)
        # ...
    # ...
```

See the other usages of the helper. The whole test block is within the context manager in the following tests:

* https://github.com/django/django/blob/fcd08c175787e909b3eb98f756317a07741c48dd/tests/signing/tests.py#L134
* https://github.com/django/django/blob/fcd08c175787e909b3eb98f756317a07741c48dd/tests/signed_cookies_tests/tests.py#L48

**Comment by timgraham** [comment]
Looks okay to me. ;-)
merged in b17a9150a0c3a132e82b53755ede62a45f897875, thanks!

**Pull request closed without merging**